brave/brave-core
# Reduce balance calls

Resolves https://github.com/brave/brave-browser/issues/9132

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
plan 1:
- enable rewards
- make sure that when you open panel / settings page no balance call is made to the server
- claim grant
- make sure that balance is shown correctly, but still no calls to the server
- restore wallet that has user funds
- make sure that balance calls are now done when you open panel / settings page
- restore wallet that has 0 user funds
- make sure that when you open panel / settings page no balance call is made to the server

plan 2:
- on master
- enable rewards
- claim grant
- close browser
- switch to this PR 
- make sure that you see only one balance call to the server, after that we shouldn't have any more balance calls

plan 3:
- on master
- enable rewards
- restore wallet with user funds
- close browser
- switch to this PR
- make sure balance calls are still done when you open panel/settings page

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
